src/SkeletalTrapezoidation.h

```diff
@@ -100,6 +101,7 @@ class SkeletalTrapezoidation
                            AngleRadians transitioning_angle
     , coord_t discretization_step_size = MM2INT(0.8)
     , coord_t transition_filter_dist = MM2INT(1)
+    , coord_t allowed_filter_deviation = MM2INT(0.2)
```

Comment:
The same could be said for the parameters before that.

I just introduced a default for it because C++ strictly enforces that all parameters with defaults should be at the end of the parameter list. I put it here in the parameter list because it's also about filtering. Therefore I had to make a default value for it.

Want me to remove all the default values in this definition?

Passing settings objects around means that code is not reusable for different purposes. It also obscures which settings are used by the class.

Reply:
True! But the parameters before that are out of scope for this issue. We can't fix all of the mistakes made in the past.

Passing settings objects around indeed has the disadvantage that the code can't be re-used for different purposes _if those different purposes need to use a different set of settings_, such as for the infill pattern generation. Currently all of the wall generation uses the same few variable line width settings (e.g. the extra infill walls still use the same transition filter setting, even if they are infill instead of walls). But it has the big advantage that the caller of the function doesn't need to change when the callee needs a new setting, like what happened here. It's better separation of concerns. I think obscuring which settings are used by a class is a good thing, since it's an implementation detail that the caller doesn't want to know about. This is pretty much why we have these settings objects instead of passing 592 parameters around.

Reply:
I'd rather fix mistakes of the past than introduce a suboptimal parameter order. I'll take a look at changing the code and let's discuss from there.

The way I see it there are classes/functions which are specific to a certain print feature and there are general classes. We wouldn't pass the settings to the Polygons class so that it can retrieve the resolution settings from there, because the Polygons class should be general. The way I see it, the SkeletelTrapezoidation class is an alternative to Polygons.offset(.), and so it should remain general. If we pass settings objects to it, it will become very difficult to use the skeleton for possible purposes in the future, such as for combing, thin feature detection, skin line direction optimization, etc.

I think this discussion shouldn't block the merging of the PR, but it's still worth having.

Reply:
Also I don't have a very fierce opinion on this, so if you stand your ground I could change the code in this respect however you prefer.